src/main/java/net/masterthought/cucumber/ReportBuilder.java

```diff
@@ -149,34 +149,31 @@ private void copyResources(String resourceLocation, String... resources) {
         }
     }
 
-    private List<AbstractPage> collectPages(Trends trends) {
-        List<AbstractPage> pages = new ArrayList<>();
-
-        pages.add(new FeaturesOverviewPage(reportResult, configuration));
-        for (Feature feature : reportResult.getAllFeatures()) {
-            pages.add(new FeatureReportPage(reportResult, configuration, feature));
-        }
-
-        pages.add(new TagsOverviewPage(reportResult, configuration));
-        for (TagObject tagObject : reportResult.getAllTags()) {
-            pages.add(new TagReportPage(reportResult, configuration, tagObject));
-        }
-
-        pages.add(new StepsOverviewPage(reportResult, configuration));
-        pages.add(new FailuresOverviewPage(reportResult, configuration));
-
-        if (configuration.isTrendsStatsFile()) {
-            pages.add(new TrendsOverviewPage(reportResult, configuration, trends));
-        }
-
-        return pages;
-    }
-
-    private void generatePages(List<AbstractPage> pages) {
-        for (AbstractPage page : pages) {
-            page.generatePage();
-        }
-    }
+	private int generatePages(Trends trends) {
+		int numPages = 0;
+		numPages += generatePage(new FeaturesOverviewPage(reportResult, configuration));
+		for (Feature feature : reportResult.getAllFeatures()) {
+			numPages += generatePage(new FeatureReportPage(reportResult, configuration, feature));
+		}
+
+		numPages += generatePage(new TagsOverviewPage(reportResult, configuration));
+		for (TagObject tagObject : reportResult.getAllTags()) {
+			numPages += generatePage(new TagReportPage(reportResult, configuration, tagObject));
+		}
+
+		numPages += generatePage(new StepsOverviewPage(reportResult, configuration));
+		numPages += generatePage(new FailuresOverviewPage(reportResult, configuration));
+
+		if (configuration.isTrendsStatsFile()) {
+			numPages += generatePage(new TrendsOverviewPage(reportResult, configuration, trends));
+		}
+		return numPages;
+	}
+	
+	private int generatePage(AbstractPage page) {
+		page.generatePage();
+		return 1;
```

Comment:
also method that returns constant value is part of code smell and should hot happen